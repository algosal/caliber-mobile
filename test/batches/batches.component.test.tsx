import React from 'react';
import 'react-native';
import 'jest-enzyme';
import '@testing-library/jest-dom';
import axios from 'axios';
import Enzyme, { shallow } from 'enzyme';

import BatchesComponent from '../../batches/batches.component';

describe('BatchesComponent Test(s)', () => {
	let trainerEmail: string =
		'mock1027.employee74df14df-5842-4811-a57c-be9836537a40@mock.com';
	let gatewayURI: string = '';

	test('the component did mount', () => {
		expect(shallow(<BatchesComponent />)).toMatchSnapshot();
	});
	test('GET Batches from API Gateway', async () => {
		//test goes here

		let returnValues;
		let obj = { data: [] };

		axios.get = jest.fn().mockResolvedValue(obj);

		// or you could use the following depending on your use case:
		// axios.get.mockImplementation(() => Promise.resolve(resp))
		// we want to test the api gateway that calls the lambda. the lambda will create multiple axios requests. we are mocking the front end request to the api gateway.

		await getBatches(trainerEmail).then((data: any) => (returnValues = data));
		//we will make one axios request in the front end to the API Gateway
		expect(axios.get).toHaveBeenCalledTimes(1);
		expect(returnValues).toBe(obj.data);
		expect(axios.get).toHaveBeenCalledWith(
			`${gatewayURI}/batches?trainerEmail=${trainerEmail}`
		);
	});
});
