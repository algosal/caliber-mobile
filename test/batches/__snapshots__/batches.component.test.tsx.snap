// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BatchesComponent Test(s) the component did mount 1`] = `
<View>
  <Text>
    BatchesComponent Placeholder
  </Text>
</View>
`;
